core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestSupport.java

```diff
@@ -513,20 +513,20 @@ public void addListener(TestListener listener) {
     private void handleApplicationPropertiesChange() {
         SmallRyeConfig updatedConfig = getMinimalConfig();
 
-        List<String> includeTags = getTrimmedListFromConfig(updatedConfig, "quarkus.test.include-tags").orElse(List.of());
-        List<String> excludeTags = getTrimmedListFromConfig(updatedConfig, "quarkus.test.exclude-tags").orElse(List.of());
+        List<String> includeTags = getTrimmedListFromConfig(updatedConfig, "quarkus.test.include-tags").orElse(null);
+        List<String> excludeTags = getTrimmedListFromConfig(updatedConfig, "quarkus.test.exclude-tags").orElse(null);
         String includePattern = updatedConfig.getOptionalValue("quarkus.test.include-pattern", String.class).orElse(null);
         String excludePattern = updatedConfig.getOptionalValue("quarkus.test.exclude-pattern", String.class).orElse(null);
-        List<String> includeEngines = getTrimmedListFromConfig(updatedConfig, "quarkus.test.include-engines").orElse(List.of());
-        List<String> excludeEngines = getTrimmedListFromConfig(updatedConfig, "quarkus.test.exclude-engines").orElse(List.of());
+        List<String> includeEngines = getTrimmedListFromConfig(updatedConfig, "quarkus.test.include-engines").orElse(null);
+        List<String> excludeEngines = getTrimmedListFromConfig(updatedConfig, "quarkus.test.exclude-engines").orElse(null);
         TestType testType = updatedConfig.getOptionalValue("quarkus.test.type", TestType.class).orElse(TestType.ALL);
 
         if (!firstRun) {
             if (!Objects.equals(includeTags, appPropertiesIncludeTags)) {
-                this.includeTags = includeTags;
+                this.includeTags = Objects.requireNonNullElse(includeTags, Collections.emptyList());
```

Comment:
Merged it but in the future, let's use `List.of()` for empty lists (no need to update it here, it's just a comment for future patches). Thanks!